### PR TITLE
add en- and decoding to municipality routing

### LINKED ISSUE #469: "Köping" municipality link from map broken
Försöker man klicka på Köping i kartan på förstasidan så kommer man till en sida med felaktig url (felaktig encoding) antagligen samma problem med alla kommuner med åäö i sitt namn?

### PR DESCRIPTION
**Pull request by elvbom**
Add en- and decoding to municipality routing as well as minor cleanup

Fix #469

### CODE REVIEW

File: __tests__/utils/generateMunipacitySitemap.test.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { TFunction } from 'next-i18next'
 import {
-  generateMunipacitySitemapData,
+  generateMunicipacitySitemapData,
```
Comment (Greenheart) [inline __tests__/utils/generateMunipacitySitemap.test.ts:3]:
```suggestion
  generateMunicipalitySitemapData,
```

File: __tests__/utils/generateMunipacitySitemap.test.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ const t = vi.fn((str) => str) as unknown as TFunction
 
 describe('generateSitemap', () => {
   it('should generate valid municipality sitemap data', () => {
-    const siteMap = generateMunipacitySitemapData({ municipalities })
+    const siteMap = generateMunicipacitySitemapData({ municipalities })
```
Comment (Greenheart) [inline __tests__/utils/generateMunipacitySitemap.test.ts:14]:
```suggestion
    const siteMap = generateMunicipalitySitemapData({ municipalities })
```

File: __tests__/utils/generateMunipacitySitemap.test.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ describe('generateSitemap', () => {
   })
 
   it('should generate a valid sitemap XML string', () => {
-    const siteMap = generateMunipacitySitemapData({ municipalities })
+    const siteMap = generateMunicipacitySitemapData({ municipalities })
```
Comment (Greenheart) [inline __tests__/utils/generateMunipacitySitemap.test.ts:34]:
```suggestion
    const siteMap = generateMunicipalitySitemapData({ municipalities })
```

File: pages/sitemap.xml.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { GetServerSideProps } from 'next'
 
 import { ClimateDataService } from '../utils/climateDataService'
 import {
-  generateMunipacitySitemapData,
+  generateMunicipacitySitemapData,
```
Comment (Greenheart) [inline pages/sitemap.xml.ts:5]:
```suggestion
  generateMunicipalitySitemapData,
```

File: pages/sitemap.xml.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ export const getServerSideProps: GetServerSideProps = async ({ res, locale }) =>
   const { t } = await getServerSideI18n(locale as string, ['common', 'sitemap'])
 
   const municipalities = new ClimateDataService().getMunicipalities()
-  const municipalitiesSitemap = generateMunipacitySitemapData({ municipalities })
+  const municipalitiesSitemap = generateMunicipacitySitemapData({ municipalities })
```
Comment (Greenheart) [inline pages/sitemap.xml.ts:14]:
```suggestion
  const municipalitiesSitemap = generateMunicipalitySitemapData({ municipalities })
```

File: utils/generateMunipacitySitemap.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ type SiteMap = {
 }
 const BASE_URL = 'https://klimatkollen.se'
 
-export const generateMunipacitySitemapData = ({
+export const generateMunicipacitySitemapData = ({
```
Comment (Greenheart) [inline utils/generateMunipacitySitemap.ts:15]:
```suggestion
export const generateMunicipalitySitemapData = ({
```

**Review by Greenheart — CHANGES_REQUESTED** [review]
One typo bit apart from that it looks good.

I will test properly in a moment.

**Review by Greenheart — APPROVED** [review]
Tested and this is a confirmed solution for #469 :)

**Comment by LudwikJaniuk** [comment]
"munipacility" <3 I think we should just make that the correct term...

**Pull request merged**